crates/router/src/core/payments.rs

```diff
@@ -3713,8 +3721,53 @@ pub async fn decide_multiplex_connector_for_normal_or_recurring_payment<F: Clone
                     .as_ref()
                     .ok_or(errors::ApiErrorResponse::InternalServerError)
                     .attach_printable("Failed to find the merchant connector id")?;
+                if is_network_token_with_transaction_id_flow(
+                    state,
+                    connector_data.connector_name,
+                    is_connector_agnostic_mit_enabled,
+                    is_network_tokenization_enabled,
+                    payment_method_info,
+                ) {
+                    if let Ok((token_exp_month, token_exp_year)) =
+                        do_status_check_for_network_token(state, key_store, payment_method_info)
+                            .await
+                    {
+                        logger::info!(
+                            "using network_tokenization with network_transaction_id for MIT flow"
+                        );
+
+                        let network_transaction_id = payment_method_info
+                            .network_transaction_id
+                            .as_ref()
+                            .ok_or(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
Add context messages using `attach_printable` here and elsewhere internal server errors are being raised.